### ISSUE TITLE
Trying to change music to custom music.

### ISSUE THREAD

**Issue by Nazkas2** [BEEmod/BEE2-items#1528]
I have my map almost finished in bee2.4, but now I want to change the default music (the one I have chosen on Bee2.4) to custom music but, when I open it in Hammer, I cant find the ambient_generic entity that plays the global music. Which instance contains this entity?

My hammer level is not very high, I've only made a couple of csgo maps

**Comment by TheDarkBomber** [comment]
Are you exporting via `puzzlemaker_export`? If you are, then that might be the problem.

**Comment by Luke18033** [comment]
Yes, `puzzlemaker_export` doesn't work with BEE2 maps - things like styles, vactubes, and music will be missing.

So you just use `ambient_generic` for music? (I'm working on a hammer map right now.)

**Comment by Nazkas2** [comment]
All I wanted to do was to find the ambient_generic and change the music it plays keeping the settings, since I dont know how to create a ambient_generic that plays music in the whole map, and not in a radial way.

**Comment by Luke18033** [comment]
As far as I know, there's no way to play a sound throughout the whole map. The `ambient_generic` is probably parented to the player.

**Comment by Luke18033** [comment]
If you're making custom music, you should really create a custom BEE2 package with the music in it. That way you won't have to edit in Hammer to change it. Also see [WASP](https://github.com/TDBapps/WASP).

**Comment by TeamSpen210** [comment]
It's a flag on the entity. The music `ambient_generic` is near the entry elevator, not in the instance (in the actual map).

**Comment by Konclan** [comment]
I would like a tutorial on how to change the music in a peti or beemod export if anyone can provide

**Comment by Nazkas2** [comment]
> It's a flag on the entity. The music ambient_generic is near the entry elevator, not in the instance (in the actual map).

When I open my map on Hammer, all I can see is the part I have built, and 3 instances: elevator_entrance, elevator_exit and arrival_departure_transition_ents. I can't see it in any of these.

> If you're making custom music, you should really create a custom BEE2 package with the music in it. That way you won't have to edit in Hammer to change it. Also see WASP.

I'm supposed to know how this WASP works? Anyway I'll ask some help from my brother.

**Comment by TheDarkBomber** [comment]
I'm creating WASP

**Comment by Byzarru** [comment]
If you want to add custom music in BEE, just make a copy of the p1music package and replace the p1 tracks with your own. Make sure to edit info.txt.

**Comment by Nazkas2** [comment]
> If you want to add custom music in BEE, just make a copy of the p1music package and replace the p1 tracks with your own. Make sure to edit info.txt.

Okay, I took the p1music package and I went to the file where the actual mp3 music is located, I pasted the custom song and, instead of editing the info.txt, I just renamed it like one of the songs that were already there. The song plays on my map, but there are two problems:

   - The song does not play in loop, once it ends, everything is silent.
   - When you interact with the gels and other things, they play the sounds assigned to the song that you have replaced (I will try what happens when replacing a song that does not have Propulsion Gel SFX, Repulsion Gel SFX, Excursion Funnel Music or Synced Funnel Music)

**Comment by TeamSpen210** [comment]
You need to edit info.txt. For the mp3s, it needs to know the length of the song. That's also where you set the filenames for the audio. Unzip the zip to a folder, then edit info.txt and change the names.

**Comment by Byzarru** [comment]
Is there a major difference between the PR23 and PR24 code?

**Comment by TeamSpen210** [comment]
Pretty much every release I've [changed](https://github.com/BEEmod/BEE2.4/pull/489) a fair bit. I added a new system for items, so they can partially reuse configurations from other styles. That would cause crashes if earlier versions loaded the new packages. There's also new code for the coop droppers, cube colourisers, and the splash screen.

**Comment by Nazkas2** [comment]
> -The song does not play in loop, once it ends, everything is silent.
-When you interact with the gels and other things, they play the sounds assigned to the song that you have replaced (I will try what happens when replacing a song that does not have Propulsion Gel SFX, Repulsion Gel SFX, Excursion Funnel Music or Synced Funnel Music).

Ok, I do not have any programming skills (except for a subject I had in college), but this is what I did to fix it: I went to the info.txt file and eddited all the lanes relating to those SFX, so I left it like this: 

"tBeam" ""
"BounceGel"
{
"snd" ""
"snd" ""
"snd" ""
"snd" ""
"snd" ""
}
"SpeedGel"
{
"snd" ""
"snd" ""
"snd" ""
"snd" ""
}
}
}

And now it wont play any sound when you are using gels or funnels. Here is the result: http://steamcommunity.com/sharedfiles/filedetails/?id=856056252

**Comment by TeamSpen210** [comment]
Just remove those sections...

Actually remove that block, and just have `"SoundScript" "file/path"`.
